app/main.py

```diff
@@ -0,0 +1,48 @@
+# from pydantic import BaseModel
+from fastapi import Depends, FastAPI, Request
+from fastapi.responses import HTMLResponse
+from fastapi.staticfiles import StaticFiles
+from fastapi.templating import Jinja2Templates
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from .models import Base, Product, SessionLocal, engine
+
+app = FastAPI()
+templates = Jinja2Templates(directory="templates")
+app.mount("/static", StaticFiles(directory="static"), name="static")
+
+@app.on_event("startup")
+def startup():
+    Base.metadata.create_all(bind=engine)
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+@app.get("/", response_class = HTMLResponse)
+async def home(request: Request, db: Session = Depends(get_db)):
```

Comment:
I noticed you are missing return annotations (`-> SomeType`) in your functions signatures. They (return annotations) are really useful when the code base starts getting bigger: you won't guess what _this very executable_ should return because it will be documented properly!
And this very function is a good starting point to start writing those. Because you'd need to understand what `templates.TemplateResponse(..)` really returns.